homeassistant/components/homewizard/switch.py

```diff
@@ -0,0 +1,147 @@
+
+from .const import COORDINATOR, DOMAIN, DeviceResponseEntry
+from .coordinator import HWEnergyDeviceUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up switches."""
+
+    coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        COORDINATOR
+    ]
+
+    if coordinator.api.state:
+        async_add_entities(
+            [
+                HWEnergyMainSwitchEntity(coordinator, entry),
+                HWEnergySwitchLockEntity(coordinator, entry),
+            ]
+        )
+
+
+class HWEnergySwitchEntity(CoordinatorEntity[DeviceResponseEntry], SwitchEntity):
+    """Representation switchable entity."""
+
+    def __init__(
+        self,
+        coordinator: HWEnergyDeviceUpdateCoordinator,
+        entry: ConfigEntry,
+        key: str,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator)
+        self.entry = entry
+
+        # Config attributes
+        self.api = coordinator.api
+        self._attr_unique_id = f"{entry.unique_id}_{key}"
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return information about the device."""
+        return {
+            "name": self.entry.title,
+            "manufacturer": "HomeWizard",
+            "sw_version": self.data["device"].firmware_version,
+            "model": self.data["device"].product_type,
+            "identifiers": {(DOMAIN, self.data["device"].serial)},
+        }
+
+    @property
+    def data(self) -> DeviceResponseEntry:
+        """Return data from DataUpdateCoordinator."""
+        return self.coordinator.data
+
+
+class HWEnergyMainSwitchEntity(HWEnergySwitchEntity):
+    """Representation of the main power switch."""
+
+    def __init__(
+        self, coordinator: HWEnergyDeviceUpdateCoordinator, entry: ConfigEntry
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator, entry, "power_on")
+
+        # Config attributes
+        self._attr_name = entry.title
+        self._attr_device_class = DEVICE_CLASS_OUTLET
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the switch on."""
+        await self.api.state.set(power_on=True)
+        await self.coordinator.async_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the switch off."""
+        await self.api.state.set(power_on=False)
+        await self.coordinator.async_refresh()
+
+    @property
+    def available(self) -> bool:
+        """
+        Return availability of power_on.
+
+        This switch becomes unavailable when switch_lock is enabled.
+        """
+        return super().available and not self.api.state.switch_lock
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if switch is on."""
+        return bool(self.api.state.power_on)
```

Comment:
Why is this converted to a bool? Isn't it a boolean?

Reply:
`self.coordinator.api.state.power_on` is an optional bool, as `state` is optional. `State` is already checked to not be None by the setup. Can we do this better?

Reply:
> is an optional bool

Right, than it is correct now 👍